### DISCUSSION TITLE
Initial submission is full dataset - are subsequent submissions deltas only and what frequency?

### DISCUSSION THREAD

**Discussion by ericwenger-pm** [nih-cfde/published-documentation#187] [Q&A] [answered]
We see that our first c2m2 submission (first round by Mar 17th, approval by Mar 18th latest) will be a _full_ list of current projects/releases/files.

Remaining Questions about subsequent submissions:
1. When we create new releases (resulting in new projects/files, potentially updates to files) do you want us to submit the _entire_ dataset of all of our releases again? or just the _new/updated_ files/metadata?
2. After Mar 18th can we resubmit at will or is there a submission freeze or other schedule to follow? for example, final updates in time for a scheduled demo / coordination meeting.

thank you!

**Comment by ACharbonneau** [comment, marked as answer]
> Remaining Questions about subsequent submissions:
> 
> 1. When we create new releases (resulting in new projects/files, potentially updates to files) do you want us to submit the _entire_ dataset of all of our releases again? or just the _new/updated_ files/metadata?

For this first release, and likely for the rest of this calendar year (2021) the portal can only accept full datapackages as submissions. So, we can't accept diffs/deltas/ etc. This is a feature we have discussed, and will look at adding if it is something submitters are interested in.

Relatedly, they system also only works in full submissions for the other stages as well: 

- On submission, each datapackage gets a unique identifier so you can tell them apart, and you can have an arbitrary number of submissions on your account at any time. Only one of those can be 'Approved' for release at any time. Once your DCC approves a specific submission for release, it sits in a queue until our next release date. If you approve a different submission before the next release, it will replace the older one in the queue. 
- On release, your most recent approved submission will be pushed to the public part of the portal, and will be your only publicly available datapackage. It does a full overwrite. (If you haven't approved a new submission, it will keep your existing data in the public site)

As of right now, none of this data is lost. All old submissions remain in your data review page, and you can see old, previously publicly available datapackages in the submission system. However in the future we will likely need to remove stale datapackages to conserve space. We also may offer public view of previous datapackages, and eventually, diff tracking to show things like data change over time if there is DCC interest.


> 2. After Mar 18th can we resubmit at will or is there a submission freeze or other schedule to follow? for example, final updates in time for a scheduled demo / coordination meeting.

Our current release schedule is quarterly, so we're planning a data release at the end of March 2021, and the next one at the end of June 2021. We will have a short (3-5 day) submission freeze the last week of March 2021 while we roll out this first release and make several bug fixes and updates we've discovered are necessary during this first use of the submission system by DCCs. However this should not affect general portal availability, only the submission tooling.
